File: 프로그래머스/lv1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.java
class Solution {
    public String[] solution(int n, int[] arr1, int[] arr2) {
        String[] answer = new String[n];

        for (int i = 0; i < n; i++) {
            String row = Integer.toBinaryString(arr1[i] | arr2[i]); // 겹친 지도의 값을 이진수로 변환
            row = String.format("%" + n + "s", row); // 지도의 크기만큼 왼쪽에 공백 추가
            row = row.replaceAll("1", "#"); // 1을 벽('#')으로 대체
            row = row.replaceAll("0", " "); // 0을 공백(' ')으로 대체
            answer[i] = row;

        }

        return answer;
    }
}

File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.py
def solution(n, arr1, arr2):
    answer = []
    for i, j in zip(arr1, arr2):
        h = bin(i | j)[2:]
        h = h.zfill(n)
        h = h.replace('1', '#')
        h = h.replace('0', ' ')
        answer.append(h)
    return answer